packages/react-components/src/gallery/dominantSpeaker.ts

```diff
@@ -3,59 +3,78 @@
 import { VideoGalleryRemoteParticipant } from '../types';
 
 /**
- * Calculates the participants that should be renderd basis on the list of dominant speakers and currently renderd participants in a call.
- * @param participants - Array containing all participants of a call. {@link @azure/communication-react#VideoGalleryRemoteParticipant}
- * @param dominantSpeakers - An array containing the userId of dominant speakers in a call in the order of their dominance. 0th index is the most dominant, 1st is the second most etc
- * @param visibleParticipants - Array containing currently rendered (visible) participants in the call. {@link @azure/communication-react#VideoGalleryRemoteParticipant}
- * @param MAX_TILES - Maximum number of tiles to calculate.
+ * Calculates the participants that should be renderd basis on the list of dominant
+ * speakers and currently renderd participants in a call.
+ * @param participants - Array containing all participants of a call.
+ * {@link @azure/communication-react#VideoGalleryRemoteParticipant}
+ * @param dominantSpeakers - An array containing the userId of dominant speakers
+ * in a call in the order of their dominance. 0th index is the most dominant, 1st is the second most etc
+ * @param visibleParticipants - Array containing currently rendered (visible)
+ * participants in the call. {@link @azure/communication-react#VideoGalleryRemoteParticipant}
+ * @param maxTiles - Maximum number of tiles to calculate.
  * @returns VideoGalleryRemoteParticipant[] {@link @azure/communication-react#VideoGalleryRemoteParticipant}
  */
 export const smartDominantSpeakerParticipants = (
   participants: VideoGalleryRemoteParticipant[] = [],
   dominantSpeakers: Array<string> = [],
   visibleParticipants: VideoGalleryRemoteParticipant[] = [],
-  MAX_TILES = 4
+  maxTiles = 4
 ): VideoGalleryRemoteParticipant[] => {
   if (!participants) return [];
 
-  // Only use the Max allowed dominant speakers.
-  const dominantSpeakerIds = dominantSpeakers.slice(0, MAX_TILES);
-
   // Don't apply any logic if total number of video streams is less than Max video streams.
-  if (participants.length <= MAX_TILES) return participants;
+  if (participants.length <= maxTiles) return participants;
 
-  // Initialize `visibleParticipants` if it is empty.
-  if (!visibleParticipants.length) {
-    visibleParticipants = participants.slice(0, MAX_TILES);
+  let currentParticipants = visibleParticipants.slice();
+  // Initialize `currentParticipants` if it is empty.
+  if (!currentParticipants.length) {
+    currentParticipants = participants.slice(0, maxTiles);
   }
 
-  const lastVisibleSpeakerIds = visibleParticipants.map((speaker) => speaker.userId);
+  // Only use the Max allowed dominant speakers.
+  const dominantSpeakerIds = Array.from(new Set(dominantSpeakers).values()).slice(0, maxTiles);
+  const lastVisibleSpeakerIds = currentParticipants.map((p) => p.userId);
```

Comment:
naming nit: Let's consistently use participants vs speakers -- speakers for stuff coming from dominant speakers, and participants for the various sub-lists from particiapnts.

So, this should be lastVisibleParticipantIDs (also, very nitty nit, I'd map from `visibleParticipants`)

Reply:
nit: consider `active` or `previously` over `last` here (last is a bit ambiguous, it could be referring to position in the array or the last known visible speakers)